### ISSUE TITLE
Docker Network Create Fails: HNS failed with error : The object already exists.

### ISSUE THREAD

**Issue by danechambers** [docker/for-win#1384]
### Expected behavior
Referencing this article: https://blogs.technet.microsoft.com/virtualization/2016/05/05/windows-container-networking/
I ran the command to create a new nat network with subnet="172.16.0.0/20". I would expect this to create another nat network.

### Actual behavior
This produced an error:  `Error response from daemon: HNS failed with error : The object already exists.`

### Information
Diagnostic id: 002AEE1F-CF19-4C9A-9033-C248AC2D6061/2017-12-05_11-29-32
```
dane@work:system32> docker info
Containers: 0
 Running: 0
 Paused: 0
 Stopped: 0
Images: 50
Server Version: 17.09.0-ce
Storage Driver: windowsfilter
 Windows:
Logging Driver: json-file
Plugins:
 Volume: local
 Network: ics l2bridge l2tunnel nat null overlay transparent
 Log: awslogs etwlogs fluentd json-file logentries splunk syslog
Swarm: inactive
Default Isolation: hyperv
Kernel Version: 10.0 16299 (16299.15.amd64fre.rs3_release.170928-1534)
Operating System: Windows 10 Pro
OSType: windows
Architecture: x86_64
CPUs: 8
Total Memory: 15.91GiB
Docker Root Dir: C:\ProgramData\Docker
Debug Mode (client): false
Debug Mode (server): true
 File Descriptors: -1
 Goroutines: 24
 System Time: 2017-12-05T11:31:39.5783892-08:00
 EventsListeners: 0
Registry: https://index.docker.io/v1/
Experimental: true
Insecure Registries:
 127.0.0.0/8
Live Restore Enabled: false
```
Logs:
>debug: RequestPool(LocalDefault, 172.16.0.0/20, , map[], false)
>debug: RequestAddress(172.16.0.0/20, <nil>, map[RequestAddressType:com.docker.network.gateway])
>debug: HNSNetwork Request ={"Name":"4639040190de7018b9ce3498c4593fd3ff0d976694a85a156c76c3851acb9736","Type":"nat","Subnets":[{"AddressPrefix":"172.16.0.0/20","GatewayAddress":"172.16.0.0"}]} Address Space=[{172.16.0.0/20 172.16.0.0 []}]
>debug: Allocating IPv4 pools for network MyNatNetwork (4639040190de7018b9ce3498c4593fd3ff0d976694a85a156c76c3851acb9736)
>debug: Calling GET /_ping
>debug: Calling POST /v1.32/networks/create
>debug: form data: {"Attachable":false,"CheckDuplicate":true,"ConfigFrom":null,"ConfigOnly":false,"Driver":"nat","EnableIPv6":false,"IPAM":{"Config":[{"Subnet":"172.16.0.0/20"}],"Driver":"default","Options":{}},"Ingress":false,"Internal":false,"Labels":{},"Name":"MyNatNetwork","Options":{},"Scope":""}
>debug: FIXME: Got an API for which error does not match any expected type!!!: HNS failed with error : The object already exists.  [module=api error_type=*errors.errorString]
>Handler for POST /v1.32/networks/create returned error: HNS failed with error : The object already exists. 
>debug: FIXME: Got an API for which error does not match any expected type!!!: HNS failed with error : The object already exists.  [module=api error_type=*errors.errorString]
>debug: ReleasePool(172.16.0.0/20)
>debug: [POST]=>[/networks/] Request : {"Name":"4639040190de7018b9ce3498c4593fd3ff0d976694a85a156c76c3851acb9736","Type":"nat","Subnets":[{"AddressPrefix":"172.16.0.0/20","GatewayAddress":"172.16.0.0"}]}
>debug: releasing IPv4 pools from network MyNatNetwork (4639040190de7018b9ce3498c4593fd3ff0d976694a85a156c76c3851acb9736)
>debug: ReleaseAddress(172.16.0.0/20, 172.16.0.0)

I did run the powershell script to clean up container networking from here: https://github.com/MicrosoftDocs/Virtualization-Documentation/tree/live/windows-server-container-tools/CleanupContainerHostNetworking

like so (from elevated powershell):
```
.\WindowsContainerNetworking-LoggingAndCleanupAide.ps1 -Cleanup -ForceDeleteAllSwitches
```
and rebooted. Still no luck.
### Steps to reproduce the behavior
  1.`docker network create -d nat --subnet="172.16.0.0/20" "MyNatNetwork"`

**Comment by jcookster** [comment]
I'm having this exact same problem. I'm thinking its a bug in docker...

**Comment by Dresel** [comment]
Having the same problem, when also specifying the gateway it works.

**Comment by lanwin** [comment]
For me it helped to simply stop the docker service. Call "Get-ContainerNetwork | Remove-ContainerNetwork" and start the docker service again.

**Comment by danechambers** [comment]
@Dresel this worked. Its definitely a little frustrating that the image from the article I referenced earlier no longer works. Then again, that article is from 2016, so I can appreciate that things have changed from then to now. At any rate, thank you for your comment! 

@lanwin I'm assuming you're referring to Get-ContainerNet from https://github.com/Microsoft/Docker-PowerShell ? I definitely appreciate that Microsoft is developing this powershell module. However, I think I'll be sticking to the docker cli to avoid any issues with the module while its the pre-release state. Thank you for your comment though!

**Comment by srikanthg10** [comment]
where is Get-ContainerNetwork  ?, can you give cli commands to remove the container network

**Comment by pangyuteng** [comment]
to elaborate abit more on @lanwin 's fix:
open powershell as admin and run the below:
```
Stop-Service docker
Get-ContainerNetwork | Remove-ContainerNetwork
Start-Service docker
```

**Comment by Cloudmersive** [comment]
Was this ever fixed?

**Comment by TomKuhn** [comment]
> Was this ever fixed?

I had this issue today, docker version 2.0.0.3, win 10 version 1803.

I could not create a new network, I had to use turn on/off windows features to remove Hyper-V and Containers, reboot a few times, put them back on, and reinstall docker.

**Comment by rbvpowell** [comment]
I was struggling with this for a while, what worked for me was:
**Get-NetNatStaticMapping** - Find the mapping with the ports I'm using.
**Remove-NetNatStaticMapping** - Remove the mapping for that port.
Then restart the hns service:
**Start-Service hns**
**Stop-Service hns**
Hope this helps.

**Comment by azmolmiah** [comment]
In the command prompt try...
`docker ps` (provides all running containers with name etc.)
`docker stop [nameofcontainer]` or `docker kill [nameofcontainer]`
You will also probably need to close other software applications such as your ide where you tried running a docker file as it may be another running container.
I was trying to install WordPress with a docker file and the one I made 5 hours ago was still running, following the above code I stopped or killed the containers, which may take a few moments by the way till the cursor moves to the next line in the terminal. Keep checking after each stop with `docker ps`. This worked for me!
